### PR TITLE
Exclude doc self

### PR DESCRIPTION
**Pull request by akatsoulas**
mozilla/sumo#1288

### CODE REVIEW

**Review by smithellis — APPROVED** [review]

**Review by escattone — CHANGES_REQUESTED** [review]
The fix for this turned out to be more extensive than I anticipated due to the caching that `_documents_for()` does. I decided to create the fix as a PR against this one:

https://github.com/akatsoulas/kitsune/pull/8

File: kitsune/wiki/facets.py [hunk 1]
```diff
@@ -69,31 +73,36 @@ def documents_for(locale, topics=None, products=None):
...
-    """Returns a list of articles that apply to passed in topics and products."""
+def _documents_for(locale, topics=None, products=None, current_document=None):
+    """Returns a list of articles that apply to passed in locale, topics and products."""
     # First try to get the results from the cache
     cache_key = _cache_key(locale, topics, products)
     documents_cache_key = f"documents_for:{cache_key}"
     documents = cache.get(documents_cache_key)
+    exclude_current_document = isinstance(current_document, Document)
     if documents is not None:
+        if exclude_current_document:
```
Comment (akatsoulas) [inline kitsune/wiki/facets.py:84]: I moved the exclusion here b/c of [this call](https://github.com/mozilla/kitsune/pull/5483/files#diff-754fad9f530ba68d2f0722f3572f6f0b45752ff296970f1839e01c660fbfd80fR63) to `_documents_for` . This way we are making sure that we are always excluding the document.

**Comment by akatsoulas** [comment]
@escattone another pass?

**Review by escattone — CHANGES_REQUESTED** [review]

File: kitsune/wiki/facets.py [hunk 2]
```diff
@@ -69,31 +73,36 @@ def documents_for(locale, topics=None, products=None):
...
-        # speed up query by removing any ordering, since we're doing it in python:
-        .order_by()
+    qs = Document.objects.filter(
+        locale=locale,
+        is_archived=False,
+        current_revision__isnull=False,
+        category__in=settings.IA_DEFAULT_CATEGORIES,
     )
+    if exclude_current_document:
+        qs = qs.exclude(id=current_document.id)
```
[the remark this answers is not in the document]
Comment (akatsoulas) [reply kitsune/wiki/facets.py:95]: My main comment here and what I tried to do in the other PR is that we need to exclude the parent in all occasions, not just when the local is the same. Any reference and any relationship to the current_document should be excluded from the fallback_documents. I am going to clean up the PR following your pattern here and we can have a final pass

**Comment by akatsoulas** [comment]
@escattone I squashed all of your work and mine in the second commit. I added a 3rd commit which is basically new code that you have not review. If this looks good I will squash the 3rd commit into the 2nd and this should be good to go. Final pass on that 3rd commit?

**Review by escattone — APPROVED** [review]
Looks great, thanks @akatsoulas!

**Pull request merged**